microcloud/mdns/mdns.go

```diff
@@ -48,3 +50,42 @@ func dnsTXTSlice(list []byte) []string {
 
 	return parts
 }
+
+// checkIPStatus checks if the interface is up, has multicast, and supports IPv4/IPv6.
+func checkIPStatus(iface string) (ipv4OK bool, ipv6OK bool, err error) {
+	netInterface, err := net.InterfaceByName(iface)
+	if err != nil {
+		return false, false, err
+	}
+
+	if netInterface.Flags&net.FlagUp != net.FlagUp {
+		return false, false, nil
+	}
+
+	if netInterface.Flags&net.FlagMulticast != net.FlagMulticast {
+		return false, false, nil
+	}
+
+	allIPv6IfacesDisabled, err := os.ReadFile("/proc/sys/net/ipv6/conf/all/disable_ipv6")
+	if err != nil {
+		return true, false, err
+	}
+
+	newIPv6IfacesDisabled, err := os.ReadFile("/proc/sys/net/ipv6/conf/default/disable_ipv6")
+	if err != nil {
+		return true, false, err
+	}
+
+	ifaceIPv6Disabled := false
+	if iface != "" {
+		ifaceIPv6DisabledBytes, err := os.ReadFile(fmt.Sprintf("/proc/sys/net/ipv6/conf/%s/disable_ipv6", iface))
+		if err != nil {
+			return true, false, err
+		}
+
+		ifaceIPv6Disabled = strings.TrimSpace(string(ifaceIPv6DisabledBytes)) == "0"
+	}
+
+	ipv6OK = strings.TrimSpace(string(allIPv6IfacesDisabled)) == "0" && strings.TrimSpace(string(newIPv6IfacesDisabled)) == "0" && ifaceIPv6Disabled
+	return true, ipv6OK, nil
```

Comment:
The naming in this function is a bit confusing for me. For each of the checks can we have something like:

```golang
allIPv6InterfacesDisabledBytes, err := os.ReadFile("/proc/sys/net/ipv6/conf/all/disable_ipv6")
if err != nil {
    return true, false, err
}

// Check disabled is false.
allIPv6InterfacesEnabled = strings.TrimSpace(string(allIPv6InterfacesDisabledBytes)) == "0"
```

Then at the bottom we'd have:
```golang
ipv6OK = allIPv6InterfacesEnabled && newIPv6InterfacesEnabled && interfaceIPv6Enabled
```

Reply:
fixed